### ISSUE TITLE
[ERROR] Not found error sent by SonarQube server (code 404)

### ISSUE THREAD

**Issue by ppc52776** [cnescatlab/sonar-cnes-report#128]
# run from command line
$ java -jar /tmp/sonar-cnes-report-3.2.2.jar -p COAMDM -s http://xxxx:9000 -t xxx
SonarQube URL: http://xxx:9000
SonarQube online: true
Detected SonarQube version: 8.2.0
```
[ERROR] Not found error sent by SonarQube server (code 404, URL http://xxx:9000/api/navigation/component?component=xxx&branch=%, Error {"errors":[{"msg":"Component key 'xxx' not found"}]}), please check cnesreport compatibility with your SonarQube server version.
```
# run from web
SonarQube web.log:
```
2020.03.29 02:53:58 ERROR web[AXEkMOydqoB87jJmAABJ][o.s.s.w.WebServiceEngine] Fail to process request http://zzz:9000/api/cnesreport/report?key=zzz&author=Administrator&token=zzz&generation=Generate
fr.cnes.sonar.report.exceptions.SonarQubeException: **Impossible to reach SonarQube instance.**
        at fr.cnes.sonar.report.ReportCommandLine.execute(ReportCommandLine.java:124)
        at fr.cnes.sonar.plugin.ws.ExportTask.handle(ExportTask.java:81)
        at org.sonar.server.ws.WebServiceEngine.execute(WebServiceEngine.java:110)
        at org.sonar.server.platform.web.WebServiceFilter.doFilter(WebServiceFilter.java:88)
        at org.sonar.server.platform.web.MasterServletFilter$GodFilterChain.doFilter(MasterServletFilter.java:139)
        at org.sonar.server.platform.web.MasterServletFilter.doFilter(MasterServletFilter.java:108)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.sonar.server.platform.web.UserSessionFilter.doFilter(UserSessionFilter.java:88)
        at org.sonar.server.platform.web.UserSessionFilter.doFilter(UserSessionFilter.java:72)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.sonar.server.platform.web.CacheControlFilter.doFilter(CacheControlFilter.java:76)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.sonar.server.platform.web.SecurityServletFilter.doHttpFilter(SecurityServletFilter.java:76)
        at org.sonar.server.platform.web.SecurityServletFilter.doFilter(SecurityServletFilter.java:48)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.sonar.server.platform.web.RedirectFilter.doFilter(RedirectFilter.java:58)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.sonar.server.platform.web.RequestIdFilter.doFilter(RequestIdFilter.java:66)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.sonar.server.platform.web.RootFilter.doFilter(RootFilter.java:62)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.filters.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:109)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
        at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:256)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:808)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by Sancretor** [comment]
Hi @ppc52776 !

Thank you for your feedback, we appreciate it !
We will look into it as soon as possible, sorry for the expected delay...

**Comment by mantri-govind** [comment]
Facing the same issue with Sonarqube 8.2 Community + Sonar-cnes 3.2.2.

**Comment by aanudev** [comment]
> Facing the same issue with Sonarqube 8.2 Community + Sonar-cnes 3.2.2.
> 
> ---- Update----
> 
> After adding sonarqube URL it started to work fine.

Where did you add SonarQuube URL to?

**Comment by Sancretor** [comment]
Just to let you know that we just released a new version compatible with 8.9 LTS.
We completely changed the way the plugin handles API calls.
It now uses internal SonarQube classes instead of localhost http calls, so it may resolve your issue.

Please let us know if it doesn't.

**Comment by RaghuCPatel** [comment]
Same issue Facing in Community 9.8 and 9.9 LTS versions
![image](https://github.com/cnescatlab/sonar-cnes-report/assets/104488667/4b743e59-b969-408d-9dd0-29dbb85f3f56)
